tools/actions_local_runner.py

```diff
@@ -70,65 +78,199 @@ def find_changed_files() -> List[str]:
     return [x.strip() for x in all_files if x.strip() != ""]
 
 
-async def run_step(step: Dict[str, Any], job_name: str, files: Optional[List[str]], quiet: bool) -> bool:
+def print_results(job_name: str, passed: bool, streams: List[str]) -> None:
+    header(job_name, passed)
+    for stream in streams:
+        stream = stream.strip()
+        if stream != "":
+            print(stream)
+
+
+async def shell_cmd(
+    cmd: Union[str, List[str]],
+    env: Optional[Dict[str, Any]] = None,
+    redirect: bool = True,
+) -> Tuple[bool, str, str]:
+    if isinstance(cmd, list):
+        cmd_str = shlex.join(cmd)  # type: ignore[attr-defined]
+    else:
+        cmd_str = cmd
+
+    proc = await asyncio.create_subprocess_shell(
+        cmd_str,
+        shell=True,
+        cwd=REPO_ROOT,
+        env=env,
+        stdout=subprocess.PIPE if redirect else None,
+        stderr=subprocess.PIPE if redirect else None,
+        executable=shutil.which("bash"),
+    )
+    stdout, stderr = await proc.communicate()
+
+    passed = proc.returncode == 0
+    if not redirect:
+        return passed, "", ""
+
+    return passed, stdout.decode().strip(), stderr.decode().strip()
+
+
+def header(name: str, passed: bool) -> None:
+    PASS = color(col.GREEN, "✓")
+    FAIL = color(col.RED, "x")
+    icon = PASS if passed else FAIL
+    print(f"{icon} {color(col.BLUE, name)}")
+
+
+def get_flake_excludes() -> List[str]:
+    config = configparser.ConfigParser()
+    config.read(os.path.join(REPO_ROOT, ".flake8"))
+
+    excludes = re.split(r',\s*', config["flake8"]["exclude"].strip())
+    excludes = [e.strip() for e in excludes if e.strip() != ""]
```

Comment:
is this line necessary?
```suggestion
```

Reply:
I'm pretty sure this is technically valid:

```
    ./caffe2,
    ,
    ./docs/caffe2,
```

which configparser pulls out as an empty string, so this just gets rid of those just in case